File: plugin/src/grokkery/Perspective.java
package grokkery;

import org.eclipse.ui.IPageLayout;
import org.eclipse.ui.IPerspectiveFactory;
import org.eclipse.ui.IPlaceholderFolderLayout;
import org.eclipse.ui.console.IConsoleConstants;

public class Perspective implements IPerspectiveFactory
{
    
    public static String id()
    {
        return "grokkery.perspective";
    }
    
    @Override
    public void createInitialLayout(IPageLayout layout)
    {
        String editorArea = layout.getEditorArea();
        layout.setEditorAreaVisible(false);
        
        IPlaceholderFolderLayout north = layout.createPlaceholderFolder("north", IPageLayout.TOP, 0.62f, editorArea);
        north.addPlaceholder("grokkery.GraphView:*");
        
        layout.addView(IConsoleConstants.ID_CONSOLE_VIEW, IPageLayout.BOTTOM, 0.62f, editorArea);
        layout.getViewLayout(IConsoleConstants.ID_CONSOLE_VIEW).setCloseable(false);
    }
    
}

File: plugin/src/grokkery/Perspective.clj
(ns grokkery.Perspective
  (:require
    [grokkery.FigureView :as FigureView])
  (:use
    clojure.contrib.import-static)
  (:gen-class
    :implements [org.eclipse.ui.IPerspectiveFactory]))

(import-static org.eclipse.ui.IPageLayout LEFT RIGHT TOP BOTTOM ID_EDITOR_AREA)
(import-static org.eclipse.ui.console.IConsoleConstants ID_CONSOLE_VIEW)


(def id "grokkery.Perspective")


(defn set-uncloseable [layout view-id]
  (.. layout (getViewLayout view-id) (setCloseable false)))


(defn -createInitialLayout [this layout]
  (.setEditorAreaVisible layout false)
  (.addPlaceholder
    (.createPlaceholderFolder layout "north" TOP (float 0.62) ID_EDITOR_AREA)
    (str FigureView/id ":*"))
  
  (.addView layout ID_CONSOLE_VIEW BOTTOM (float 0.62) ID_EDITOR_AREA)
  (set-uncloseable layout ID_CONSOLE_VIEW))    